fs/fs.go

```diff
@@ -287,6 +297,9 @@ func (c *sociContext) Init(ctx context.Context, imageRef, indexDigest, imageMani
 		c.sociIndex = index
 		c.populateImageLayerToSociMapping(index)
 	})
+	c.cachedErrMu.Lock()
+	retErr = c.cachedErr
+	c.cachedErrMu.Unlock()
```

Comment:
This could be an Rlock on a RWMutex to allow simultaneous reads. This is not a blocking review, feel free to resolve and merge if nothing else is in the way and you aren't pushing any further changes otherwise.